wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginMetaRetriever.java

```diff
@@ -87,11 +93,23 @@ public PluginMetaRetriever(@Nullable @Named(PLUGIN_REGISTRY_PROPERTY) String plu
    */
   @Beta
   public Collection<PluginMeta> get(Map<String, String> attributes) throws InfrastructureException {
-    if (pluginRegistry == null || attributes == null || attributes.isEmpty()) {
+    // Have to check for empty value instead of plain null as it's possible to have empty
+    // plugins attribute
+    String pluginsAttribute =
+        attributes.getOrDefault(Constants.WORKSPACE_TOOLING_PLUGINS_ATTRIBUTE, null);
+    String editorAttribute =
+        attributes.getOrDefault(Constants.WORKSPACE_TOOLING_EDITOR_ATTRIBUTE, null);
+
+    // Check if workspace requests ws.next features with no registry configured
+    if (pluginRegistry == null
+        && (!Strings.isNullOrEmpty(pluginsAttribute) || !Strings.isNullOrEmpty(editorAttribute))) {
+      throw new InfrastructureException(CHE_REGISTRY_MISSING_ERROR);
```

Comment:
should a test be added ?

Reply:
I've added basic test cases to cover backwards compatability (no registry / no ws.next features).